packages/authentication/src/decorators/authenticate.decorator.ts

```diff
@@ -52,14 +52,26 @@ export function authenticate(
 
     if (method && methodDescriptor) {
       // Method
+      MethodDecoratorFactory.createDecorator<SecurityRequirementObject[]>(
+        OAI3Keys.SECURITY_METHOD_KEY,
+        specs.map(o => ({[o.strategy]: []})),
```

Comment:
@frbuceta one problem I see with this approach is that if a strategy does not define a `securityScheme` the openapi spec will be invalid

Reply:
Well seen, I had thought about it when I started to program it but I had forgotten. I'm going to see what solutions I have

Reply:
Maybe another approach could be to utilize a spec enhancer and allow authentication strategies to specify the security spec, see [gist](https://gist.github.com/nflaig/d8b6b30d0d328eae11eefe3e4ff50dce).